application-ideas-ui/src/main/resources/Ideas/IdeasClass.xml

```diff
@@ -508,121 +508,112 @@
               againstText.empty();
             } else {
-              $(e).addClass(voteClass);
-              if(voteClass == 'voted-pro') {
-                supportText.html("$services.localization.render('ideas.ideassheet.voted')");
-                againstText.empty();
-              } else {
-                againstText.html("$services.localization.render('ideas.ideassheet.votedagainst')");
-                supportText.empty();
-              }
-            }
-          });
-          opponent.forEach(function(e) {
-            $(e).removeClass(againstClass);
-          });
-          // Updating the votes. Refreshing the list of users that have voted with an idea.
-          if(voteClass == 'voted-pro') {
-            var nbv = response.nbvotes;
-            $('#pro-participants').load(XWiki.currentDocument.getURL('get') + ' #pro-participants&gt;*', '');
-            if(response.nbagainst || response.remove) {
-              $('#against-participants').load(XWiki.currentDocument.getURL('get') + ' #against-participants&gt;*', '');
-            }
-          } else {
-            var nbv = response.nbagainst;
-            $('#against-participants').load(XWiki.currentDocument.getURL('get') + ' #against-participants&gt;*', '');
-            if(response.nbvotes || response.remove) {
-              $('#pro-participants').load(XWiki.currentDocument.getURL('get') + ' #pro-participants&gt;*', '');
+              againstText.html("$services.localization.render('ideas.ideassheet.votedagainst')");
+              supportText.empty();
             }
           }
-          voted.forEach(function(e) {
-            $(e).html(nbv);
-          });
-          if(!refresh) {
-            location.reload();
-          }
-          else if(response.remove) {
-            var nbdec = response.nbopp;
-            opponent.forEach(function(e) {
-              $(e).html(nbdec);
-            });
-          }
+        });
+        opponent.forEach(function (e) {
+          $(e).removeClass(againstClass);
+        });
+        // Updating the votes. Refreshing the list of users that have voted with an idea.
+        if (voteClass == 'voted-pro') {
+          var nbv = response.supporters.length;
+          var nbop = response.opponents.length;
         } else {
-          new XWiki.widgets.Notification(
-            response.msg || "$services.localization.render('ideas.ideasclass.vote.error')", 'error');
+          var nbv = response.opponents.length;
+          var nbop = response.supporters.length;
+        }
+        $('#pro-participants').load(XWiki.currentDocument.getURL('get') + ' #pro-participants>*', '');
+        $('#against-participants').load(XWiki.currentDocument.getURL('get') + ' #against-participants>*', '');
+        voted.forEach(function (e) {
+          $(e).html(nbv);
+        });
+        opponent.forEach(function (e) {
+          $(e).html(nbop);
+        });
+        if (!refresh) {
+          location.reload();
```

Comment:
The variable name ``refresh`` is misleading. This reads "if not refresh then refresh", as in "if no refresh is needed then we refresh" which is awkward.